graphbolt/include/graphbolt/shared_mem.h

```diff
@@ -0,0 +1,105 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ *
+ * @file shm/shared_mem.h
+ * @brief shared memory management.
+ */
+#ifndef GRAPHBOLT_SHM_SHARED_MEM_H_
+#define GRAPHBOLT_SHM_SHARED_MEM_H_
+
+#ifdef _WIN32
+#include <windows.h>
+#endif  // _WIN32
+#include <string>
+
+namespace graphbolt {
+namespace sampling {
+
+/**
+ * @brief This class owns shared memory.
+ *
+ * When the object is gone, the shared memory will also be destroyed.
```

Comment:
This is not implemented by the SharedMemory but the unique_ptr mechanism in CSCSamplingGraph